component/azstorage/azstorage.go

```diff
@@ -386,7 +425,12 @@ func (az *AzStorage) GetFileBlockOffsets(options internal.GetFileBlockOffsetsOpt
 
 func (az *AzStorage) TruncateFile(options internal.TruncateFileOptions) error {
 	log.Trace("AzStorage::TruncateFile : %s to %d bytes", options.Name, options.Size)
-	return az.storage.TruncateFile(options.Name, options.Size)
+	err := az.storage.TruncateFile(options.Name, options.Size)
+
+	if err == nil {
+		AzStatsCollector.AddStats(az.Name(), "TruncateFile", options.Name, true, map[string]interface{}{"Size": options.Size})
```

Comment:
name fo the file being truncated as part of event

Reply:
Already have `options.Name` here which will be the relative path of the file being truncated.